app/services/efile/poll_for_acknowledgments_service.rb

```diff
@@ -89,10 +89,17 @@ def self._handle_submission_status_response(response)
       status_updates = 0
 
       doc.css('StatusRecordGrp').each do |status_record_group|
-        if status_record_group.css('SubmissionStatusTxt').text == 'Acknowledgement Received from State'
-          status_updates += 1
-          submission = EfileSubmission.find_by(irs_submission_id: status_record_group.css('SubmissionId').text)
+        status = status_record_group.css('SubmissionStatusTxt').text
+        status_updates += 1
+        submission = EfileSubmission.find_by(irs_submission_id: status_record_group.css('SubmissionId').text)
+        if status == 'Received'
           submission.transition_to(:ready_for_ack, raw_response: status_record_group.to_xml)
+        elsif status == 'Denied by IRS'
```

Comment:
We should keep track of the error status responses we get in a constant somewhere.